NewForce-at-Mountwest/bangazon-api-the-thunder-lobsters
# Product Resource

# Description

Added functionality to GET, POST, PUT, and DELETE Products from the Bangazon API. Added a test suite for integration testing of this functionality.

## Related Ticket(s)

https://trello.com/c/VCZIOXgH/2-ticket-2-allow-developers-to-access-a-product-resource

## Steps to Test

- Pull down these changes on a new branch
- Open a new SQL query for your database
- Copy the following code into the new SQL query and execute it:
`INSERT INTO ProductType (Name) VALUES ('Type One');`
`INSERT INTO ProductType (Name) VALUES ('Type Two');`
`INSERT INTO Customer (FirstName, LastName, CreationDate, LastActiveDate) VALUES ('Dylan','Bishop', '2020-08-16 11:46:23', '2020-08-16 11:46:23');`
`INSERT INTO Customer (FirstName, LastName, CreationDate, LastActiveDate) VALUES ('Lindsey', 'Crittendon', '2020-08-17 11:46:23', '2020-08-16 11:46:23');`
`INSERT INTO Customer (FirstName, LastName, CreationDate, LastActiveDate) VALUES ('Derek', 'Mayse', '2020-08-18 11:46:23', '2020-08-16 11:46:23');`
`INSERT INTO Customer (FirstName, LastName, CreationDate, LastActiveDate) VALUES ('Barry', 'Griffith', '2020-08-19 11:46:23', '2020-08-16 11:46:23');`
`INSERT INTO Customer (FirstName, LastName, CreationDate, LastActiveDate) VALUES ('Ashon', 'Woodbury', '2020-08-16 11:46:23', '2020-08-16 11:46:23');`
`INSERT INTO Customer (FirstName, LastName, CreationDate, LastActiveDate) VALUES ('Potato', 'Derek', '2020-08-15 11:46:23', '2020-08-16 11:46:23');`
`INSERT INTO Product (ProductTypeId, CustomerId, Price, Title, Description, Quantity) VALUES ((SELECT TOP 1 Id from ProductType WHERE Name = 'Type One'), (SELECT TOP 1 Id FROM Customer WHERE FirstName = 'Potato'), 1, 'Test One', 'Test One Description', 1);`
`INSERT INTO Product (ProductTypeId, CustomerId, Price, Title, Description, Quantity) VALUES ((SELECT TOP 1 Id from ProductType WHERE Name = 'Type Two'), (SELECT TOP 1 Id FROM Customer WHERE FirstName = 'Ashon'), 2, 'Test Two', 'Test Two Description', 2);`
- Edit `AppSettings.json` (in BangazonAPI) and `DatabaseFixture.cs` (in TestBangazonAPI) to reference the SQL server running on your machine, rather than SQLEXPRESS01.
- Press CTRL + F5 to start the API
- Change the URL of your browser to end in `api/products`
- Copy and paste this URL into Postman
- Using Postman, ensure that you can GET a list of all products from this URL.
- Ensure that you can add an ID to the end of the URL to GET a single product resource
- Copy JSON from your GET result and use it to POST a new product
- Use PUT to edit the product you just POSTed
- Use GET to ensure this product was edited
- DELETE the product you created
- Use GET to make sure the product is now removed
- In Visual Studio, navigate to the Test Explorer and run all tests. Ensure that all 5 tests succeed.

# Checklist:
